nautobot/dcim/tests/test_models.py

```diff
@@ -1994,3 +2001,560 @@ def test_parent_controller_change(self):
             controller2,
             "Child group 2 controller should have been updated",
         )
+
+
+class ModuleBayTestCase(ModelTestCases.BaseModelTestCase):
+    model = ModuleBay
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.device = Device.objects.first()
+        cls._delete_module_bays(cls.device)
+        cls.module = Module.objects.first()
+        cls._delete_module_bays(cls.module)
+
+    # TODO: should this be a helper method (cascade_delete?) on the ModuleBay model?
+    # This works around the relationship between ModuleBay and its installed Module being protected
```

Comment:
Huh. Don't we have `on_delete=CASCADE` on that field? Would like to understand why this is an issue.

Reply:
I had it set to PROTECT before but when I realized that every other component of device is CASCADE I changed it. I will remove this from the test